projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts

```diff
@@ -8,15 +8,21 @@ import { MatPaginator, MatPaginatorModule } from '@angular/material/paginator';
   templateUrl: './pagination.component.html',
   styleUrl: './pagination.component.scss'
 })
-export class PaginationComponent {
+export class PaginationComponent implements OnChanges{
 
   @ViewChild(MatPaginator) paginator!: MatPaginator;
   @Input() totalCount:any;
-  @Input() setPaginatorToFirstpage:any
+  @Input() setPaginatorToFirstpage:boolean = false;
   @Output() paginatorChanged = new EventEmitter();
   @Input() pageSize:any = 5
   @Input() pageSizeOptions:Array<number>= [5,10,20,100]
- 
+
+  ngOnChanges() {
+    if(this.setPaginatorToFirstpage){
```

Comment:
why do we need ngOnChanges here ?

Reply:
Issue: for example if the user is on pagination 4 and tries to search the content reflects on pagination 4
Suma refered me to use ngOnchange to fix the issue.

Reply:
we emit event right ?